stm/stm.ml

```diff
@@ -289,7 +344,8 @@ module VCS : sig
   val get_info : id -> vcs state_info
   val reached : id -> unit
   val goals : id -> int -> unit
-  val set_state : id -> cached_state -> unit
+  val set_state : id -> exec_state -> unit
+  val set_parsing_state : id -> Parser.state -> unit
   val get_state : id -> cached_state
```

Comment:
I'll add a comment, the latter one is only needed now because of the mutable thingy.